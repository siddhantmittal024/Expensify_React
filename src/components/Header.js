import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      ExpenseHelp
    </NavLink>
  </header>
);

export default Header;
