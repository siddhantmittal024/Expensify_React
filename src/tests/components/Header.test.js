//import ReactShallowRenderer from 'react-test-renderer/shallow';
import {shallow} from 'enzyme';
import React from 'react';
import Header from '../../components/Header';

test('should render Header correctly',()=>{
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
    

    //expect(wrapper.find('h1').text()).toBe('Expensify');
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();

});
