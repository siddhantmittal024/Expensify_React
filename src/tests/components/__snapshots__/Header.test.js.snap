// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <[object Object]
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/create"
  >
    Create
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/help"
  >
    ExpenseHelp
  </[object Object]>
</header>
`;
